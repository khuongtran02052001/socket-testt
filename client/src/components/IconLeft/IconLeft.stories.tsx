import React from 'react';

import { Meta, Story } from '@storybook/react';

import { IconLeft, IconProps } from './IconLeft';

const story: Meta = {
  title: 'Components/IconLeft',
  component: IconLeft,
};

export const Default: Story = (props: IconProps) => {
  return <IconLeft {...props}></IconLeft>;
};

Default.args = {};

Default.parameters = {
  jest: ['Header.test.tsx'],
  layout: 'fullscreen',
};

export default story;
