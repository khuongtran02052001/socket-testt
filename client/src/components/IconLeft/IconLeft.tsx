/* eslint-disable consistent-return */
import React from 'react';

import { useRouter } from 'next/router';

import { LOCAL_STORAGE } from '../../Context/contant';
import { toast } from 'react-toastify';

import { DataLeft } from 'data/DataLeft';
import CardLeft from 'src/card/ CardLeft';

import { motion } from 'framer-motion'
import { fadeUp, routeAnimation } from 'src/motion';

export interface IconProps {
  className?: string;
}

export const IconLeft: React.FC<IconProps> = (props) => {
  const router = useRouter();
  const handleSubmit = () => {
    if (localStorage.getItem(LOCAL_STORAGE)) {
      return toast.warning(`You need to log out !`);
    }
    router.push('/register');
  };
  const handleSubmit1 = () => {
    if (localStorage.getItem(LOCAL_STORAGE)) {
      return toast.warning(`You need to log out !`);
    }
    router.push('/login');
  };
  return (
    <div
      className="sm:hidden dark:text-black' w-full md:w-1/6  text-sm hidden"
      {...props}
    >
      <div className="p-4 bg-white rounded-lg shadow-lg">
        <span className="mx-1 font-bold text-blue-600 cursor-pointer">
          DEV Community
        </span>
        <span className="font-bold">
          is a community of 697,706 amazing developers
        </span>
        <br />
        <br />

        <span className="my-7">
          We&apos;re a place where coders share, stay up-to-date and grow their
          careers.
        </span>
        <motion.div
          variants={routeAnimation}
          initial="initial"
          animate="animate"
          exit="exit"
          onClick={handleSubmit1}
          className="p-2 my-2 text-center text-blue-600 bg-white rounded-md cursor-pointer hover:bg-gray-300"
        >
          <button>Login</button>
        </motion.div>
        <motion.div
          variants={routeAnimation}
          initial="initial"
          animate="animate"
          exit="exit"
          onClick={handleSubmit}
          className="p-2 my-5 text-center text-white bg-blue-500 rounded-md cursor-pointer"
        >
          <button>Create new account</button>
        </motion.div>
      </div>
      <motion.div
        variants={routeAnimation}
        initial="initial"
        animate="animate"
        exit="exit"
        className='block'>
        {DataLeft.map(item => (
          <motion.div
            variants={fadeUp}
            initial="initial"
            animate="animate"
          >
            <CardLeft item={item} />
          </motion.div>
        ))}
      </motion.div>
    </div>
  );
};
