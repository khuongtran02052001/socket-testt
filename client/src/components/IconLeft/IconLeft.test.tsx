import React from 'react';

import renderer from 'react-test-renderer';

import { IconLeft } from './IconLeft';

it('renders correctly', () => {
  const tree = renderer.create(<IconLeft></IconLeft>).toJSON();
  expect(tree).toMatchSnapshot();
});
