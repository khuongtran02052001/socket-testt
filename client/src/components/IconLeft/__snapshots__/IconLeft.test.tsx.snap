// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sm:block dark:text-black' w-full md:w-1/6  text-sm hidden"
>
  <div
    className="p-4 bg-white rounded-lg shadow-lg"
  >
    <span
      className="mx-1 font-bold text-blue-600 cursor-pointer"
    >
      DEV Community
    </span>
    <span
      className="font-bold"
    >
      is a community of 697,706 amazing developers
    </span>
    <br />
    <br />
    <span
      className="my-7"
    >
      We're a place where coders share, stay up-to-date and grow their careers.
    </span>
    <div
      className="p-2 my-2 text-center text-blue-600 bg-white rounded-md cursor-pointer hover:bg-gray-300"
      onClick={[Function]}
    >
      <button>
        Login
      </button>
    </div>
    <div
      className="p-2 my-5 text-center text-white bg-blue-500 rounded-md cursor-pointer"
      onClick={[Function]}
    >
      <button>
        Create new account
      </button>
    </div>
  </div>
  <div
    className="block text-center transition-all transform bg-white border-2 border-gray-400 shadow-md hover:scale-110 border-ridge my-7 "
  >
    <div
      className="relative cursor-pointer "
    >
      <img
        className="w-10 h-10 m-auto"
        src="https://logopond.com/logos/5d50b7e780862243f64312c7c282e654.png"
      />
      <p
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Home
      </p>
    </div>
  </div>
  <div
    className="block text-center transition-all transform bg-white border-2 border-gray-400 shadow-md hover:scale-110 border-ridge my-7 "
  >
    <div
      className="relative cursor-pointer "
      onClick={[Function]}
    >
      <img
        className="w-8 h-8 m-auto"
        src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRAk_4GzoANPRm6D5sKl8ZtAy8KT7mrpAeC-Q&usqp=CAU"
      />
      <p
        className=""
      >
        SignIn/SignUp
      </p>
    </div>
  </div>
  <div
    className="flex justify-center w-1/5 p-2 m-auto my-5 text-center text-white bg-black rounded-full cursor-pointer"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <button
      className="flex"
    >
      <svg
        className="w-6 h-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 13l-7 7-7-7m14-8l-7 7-7-7"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
        />
      </svg>
    </button>
  </div>
</div>
`;
